src/bin/languageserver/mod.rs

```diff
@@ -70,9 +72,12 @@ pub async fn start_server(language_args: &LanguageServerCommand) -> ! {
 impl SolangServer {
     /// Parse file
     async fn parse_file(&self, uri: Url) {
+        // if let Some(contents) = self.text_buffers.lock().await.get(&path) {
+        let mut resolver = FileResolver::new();
+        for (path, contents) in self.text_buffers.lock().await.iter() {
+            resolver.set_file_contents(path.to_str().unwrap(), contents.clone());
+        }
```

Comment:
I do wonder if it makes more sense to keep a `FileResolver` around and keep that up to date. `FileResolver` could have methods for updating the file (or cache).

Just an suggestion.

Reply:
`FileResolver` seems to be designed for the use case where the files are just added once. 
For example, certain methods return an error when the path being added already exists. Also, `FileResolver` stores the files as a `Vec` and not `HashMap` or `HashSet` which can be used to deduplicate entries. 
So, I believe some changes need to be made to `FileResolver` before this idea can be incorporated.